config/src/main/java/com/alibaba/nacos/config/server/service/ConfigDetailService.java

```diff
@@ -0,0 +1,245 @@
+                String.valueOf(getMaxThread()))), MAX_THREAD));
+        setWaitTimeout(Integer.parseInt(EnvUtil.getProperty(PropertiesConstant.SEARCH_WAIT_TIMEOUT,
+                String.valueOf(getWaitTimeout()))));
+    }
+    
+    /**
+     * init worker thread.
+     */
+    private void initWorker() {
+        this.eventLinkedBlockingQueue = new LinkedBlockingQueue<>(maxCapacity);
+        
+        clientEventExecutor = new ScheduledThreadPoolExecutor(maxThread, r -> {
+            Thread t = new Thread(r);
+            t.setName("com.alibaba.nacos.config.search.worker");
+            t.setDaemon(true);
+            return t;
+        });
+        
+        for (int i = 0; i < maxThread; i++) {
+            clientEventExecutor.submit(() -> {
+                while (true) {
+                    try {
+                        SearchEvent event = eventLinkedBlockingQueue.take();
+                        Page<ConfigInfo> result = null;
+                        if (Constants.CONFIG_SEARCH_BLUR.equals(event.getType())) {
+                            result = configInfoPersistService.findConfigInfoLike4Page(event.pageNo, event.pageSize,
+                                    event.dataId, event.group, event.tenant, event.configAdvanceInfo);
+                        } else {
+                            result = configInfoPersistService.findConfigInfo4Page(event.pageNo, event.pageSize,
+                                    event.dataId, event.group, event.tenant, event.configAdvanceInfo);
+                        }
+                        synchronized (event) {
+                            event.setResponse(result);
+                            event.notifyAll();
+                        }
+                    } catch (Exception e) {
+                        LOGGER.error("catch search worker error: {}", e.getMessage());
+                    }
+                }
+            });
+        }
+    }
+    
+    /**
+     * block thread and use workerThread to search config.
+     */
+    public Page<ConfigInfo> findConfigInfoPage(String search, int pageNo, int pageSize, String dataId, String group,
+            String tenant, Map<String, Object> configAdvanceInfo) throws NacosException {
+        SearchEvent searchEvent = new SearchEvent(search, pageNo, pageSize, dataId, group, tenant,
+                configAdvanceInfo);
+        Page<ConfigInfo> result = null;
+        try {
+            synchronized (searchEvent) {
+                boolean offer = eventLinkedBlockingQueue.offer(searchEvent);
+                if (!offer) {
+                    throw new RuntimeException("config detail event offer fail.");
+                }
+                searchEvent.wait(waitTimeout);
+                result = searchEvent.getResponse();
+            }
+        } catch (RuntimeException e) {
+            LOGGER.error("config detail block queue add error: {}.", e.getMessage());
```

Comment:
No need catch RuntimeException and change to NacosRuntimeException.

Directly throw NacosRuntimeException is ok.